src/components/search-input/SearchInput.jsx

```diff
@@ -8,7 +8,7 @@ import SearchIcon from '@mui/icons-material/Search'
 import { styles } from './SearchInput.styles'
 
 const SearchInput = ({ search, setSearch }) => {
-  const [searchInput, setSearchInput] = useState(search)
+  const [searchInput, setSearchInput] = useState('' || search)
```

Comment:
`searchInput` will always equal to `search` in this case, because an empty string is always `false`. If you want to handle the case `search===undefined` rearrange the parts: 
```suggestion
  const [searchInput, setSearchInput] = useState(search || '' )
```